addons/mail/static/src/components/discuss/tests/discuss_tests.js

```diff
@@ -4206,6 +4206,65 @@ QUnit.test('auto-focus composer on opening thread', async function (assert) {
     );
 });
 
+QUnit.test('mark channel as seen if last message is visible when switching channels [REQUIRE FOCUS]', async function (assert) {
```

Comment:
I think we also need to add "... even when the previous channel had a more recent last message than the current channel"
because this was actually the bug. In the normal case as described by current title it already worked before.

Reply:
Shouldn't we actually have 2 tests here? 1 for the focus issue, and 1 for the last message issue?

Reply:
I don't know how you could test the focus issue ? By the way, the .focus() is no more needed in this test, we could remove it

Reply:
Definitely remove the focus from this test if it's no longer needed, and update title as suggested above.

As for how to test the focus issue... Well the same way you detected it in the first place, just gotta reproduce that flux in a test.

Reply:
Yeah but the problem was in model layer, so difficult to check anything in the UI ..

Reply:
I mean the fix of focus that we do in `_update` in this PR, this one is in a component at least so there must be a way?

Reply:
in the UI nothing changes really.. in fact the both fix are needed to fix the found problem : we need the focus to be "redispatched" when switching channels AND we need the lastMessage to be bound the thread shown as otherwise the markAsSeen is not called as the shown message is older than the last notified as seen or the composer is already focused

Reply:
That means if we wanted to test the focus case we would have to write the exact same test except the thread to which we switch into has a more recent message instead of least recent? But this is the standard case, we never noticed that issue in practice I think? Was the focus issue only manifesting in the current test, or also when testing manually?

Reply:
Only in test, because when you click in real situation, the focus is also changed so the problem was not showing up, but in test you had to focus and then click ... By the way the t-key seems to work too, so maybe we don't need to test this anymore as it's strongly related to test conditions ?

Reply:
Ok to just keep this test.